### ISSUE TITLE
Autocomplete not triggering for JS/TS files

### ISSUE THREAD

**Issue by azemetre** [neoclide/coc-tsserver#171]
Hello!

After recently upgrading to `coc-tsserver` to v1.5.2 and `coc.nvim` to v0.78 I'm having issues with <tab> autocomplete triggering.

Jumping to definitions or references still works, errors will still appear for bad syntax or types. Everything about `coc-tsserver` appears to be working correctly except <tab> autocompletion.

In my vimrc `:verbose imap <tab>` is only being defined once, defined the same way as in the `coc.nvim` readme.

Other coc extensions like `coc-css` or `coc-json` work fine with <tab> autocomplete in their respective filetypes but not with JS/JSX/TS/TSX files using `coc-tsserver`.

I've reinstalled `coc.nvim` and all my coc extensions but the problem still persists.

I feel like I am missed something simple here. Is there anything I am forgetting? Anything else I can attach to help debug? I don't know how useful `tsserver.trace.server` logs would be, the issues is that a key isn't triggering a correct action correct? That won't be reflected in the logs, would it?

Is there another way to force autocompletion to test it?

I do use nvm for work but I have deactivated it and the problem still persists.

Attached are my `CocInfo` logs.

<details>
## versions

vim version: NVIM v0.4.3
node version: v12.14.1
coc.nvim version: 0.0.78-3e5a86c1f8
coc.nvim directory: /Users/azemetre/.vim/plugged/coc.nvim
term: iTerm.app
platform: darwin

## Output channel: snippets

[Info 10:47:07 PM] Using ultisnips directories: UltiSnips /Users/azemetre/.config/coc/ultisnips
[Info 10:47:07 PM] Using ultisnips python command: pyx

## Output channel: tsserver

[Info  - 10:47:07 PM] TSServer log file :
/var/folders/l7/fbvg35fj2ls6jgnfcs2v9n040000gn/T/coc.nvim-22415/tsserver.log
[Info  - 10:47:07 PM] Forking TSServer
PATH: /Users/azemetre/google-cloud-sdk/bin:/Users/azemetre/.nvm/versions/node/v12.14.1/bin:/usr/local/opt/coreutils/libexec/gnubin:/Users/azemetre/.dotnet/tools:/usr/local/sbin:/Users/azemetre/bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands 
[Info  - 10:47:07 PM] Started TSServer
{
  "path": "/Users/azemetre/company-repo/company-project/node_modules/typescript/lib",
  "_api": {
    "versionString": "3.7.2",
    "version": "3.7.2"
  }
}
[Trace  - 10:47:07 PM] Sending request: configure (0). Response expected: no. Current queue length: 0
Arguments: {
    "hostInfo": "coc-nvim",
    "preferences": {
        "providePrefixAndSuffixTextForRename": true,
        "allowRenameOfImportPath": true
    }
}
[Trace  - 10:47:07 PM] Sending request: compilerOptionsForInferredProjects (1). Response expected: no. Current queue length: 0
Arguments: {
    "options": {
        "module": "commonjs",
        "target": "es2016",
        "jsx": "preserve",
        "allowJs": true,
        "allowSyntheticDefaultImports": true,
        "allowNonTsExtensions": true
    }
}
[Trace  - 10:47:07 PM] Sending request: open (2). Response expected: no. Current queue length: 0
Arguments: {
    "file": "/Users/azemetre/company-repo/company-project/src/routes/contact-us/ContactUs.tsx",
    "fileContent": "/Users/azemetre/company-repo/company-project/src/routes/contact-us/ContactUs.tsx",
    "scriptKindName": "TSX",
    "projectRootPath": "/Users/azemetre/company-repo/company-project"
}
[Trace  - 10:47:07 PM] Sending request: configure (3). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/azemetre/company-repo/company-project/src/routes/contact-us/ContactUs.tsx",
    "formatOptions": {
        "tabSize": 2,
        "indentSize": 2,
        "convertTabsToSpaces": true,
        "newLineCharacter": "\n",
        "insertSpaceAfterCommaDelimiter": true,
        "insertSpaceAfterConstructor": false,
        "insertSpaceAfterSemicolonInForStatements": true,
        "insertSpaceBeforeAndAfterBinaryOperators": true,
        "insertSpaceAfterKeywordsInControlFlowStatements": true,
        "insertSpaceAfterFunctionKeywordForAnonymousFunctions": true,
        "insertSpaceBeforeFunctionParenthesis": false,
        "insertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis": false,
        "insertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets": false,
        "insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces": false,
        "insertSpaceAfterOpeningAndBeforeClosingTemplateStringBraces": false,
        "insertSpaceAfterOpeningAndBeforeClosingJsxExpressionBraces": false,
        "insertSpaceAfterTypeAssertion": false,
        "placeOpenBraceOnNewLineForFunctions": false,
        "placeOpenBraceOnNewLineForControlBlocks": false,
        "semicolons": "ignore"
    },
    "preferences": {
        "quotePreference": "auto",
        "importModuleSpecifierPreference": "auto",
        "importModuleSpecifierEnding": "auto",
        "allowTextChangesInNewFiles": true,
        "allowRenameOfImportPath": true,
        "providePrefixAndSuffixTextForRename": true
    }
}
[Trace  - 10:47:07 PM] Event received: typingsInstallerPid (0).
Data: {
    "pid": 22424
}
[Trace  - 10:47:07 PM] Event received: projectLoadingStart (0).
Data: {
    "projectName": "/Users/azemetre/company-repo/company-project/tsconfig.json",
    "reason": "Creating possible configured project for /Users/azemetre/company-repo/company-project/src/routes/contact-us/ContactUs.tsx to open"
}

## Output channel: highlight

[Info  - 10:47:07 PM] Highlight server running in node v12.14.1
</details>

**Comment by typkrft** [comment]
I'm having the same issue. Auto complete does not work with .js files on a brand new install of nvim, coc, tsserver. If there's any particular logs or information that would help please let me know.

**Comment by azemetre** [comment]
Sorry! Meant to reply soon with what worked for me.

I went thru the same motions @typkrft , I noticed that the version of typescript and yarn that I had installed on my global npm (I use nvm but point `tsserver.npm` in my `coc.json` to be local user as defined in the docs) was several months out of date.

After updating them, along with a fresh reinstall of coc.nvim and coc-tsserver, I was able to get it working correctly.
